File: Data Structures & Algorithms/partition-labels/submission-5.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
        int n = s.length();
        int last[26];
        for(int i = 0; i < n; ++i) last[s[i] - 'a'] = i;

        int st = 0, ed = 0;
        vector<int> ans;
        for(int i = 0; i < n; ++i) {
            ed = max(ed, last[s[i] - 'a']);
            if(i == ed) {
                ans.push_back(ed - st + 1);
                st = i + 1;
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/non-overlapping-intervals/submission-5.js
class Solution {
    /**
     * @param {number[][]} intervals
     * @return {number}
     */
    eraseOverlapIntervals(intervals) {
        intervals.sort((a, b) => a[1] - b[1]);
        let ans = 0;
        let last = -Infinity;
        for(const [st, ed] of intervals) {
            if(st >= last) last = ed;
            else ++ans;
        }
        return ans;
    }
}
